query07.sql

```diff
@@ -1,6 +1,18 @@
 /*
   How many trips started on one day and ended in the next?
+
+4,782
+
 */
 
 -- Enter your SQL query here
-select ...
+WITH daytbl as (  
+SELECT trip_id, start_time, end_time, EXTRACT(DAY FROM start_time) as start_time_day, EXTRACT (DAY FROM end_time) as end_time_day
+FROM public.indego_trips_2019_q2
+UNION ALL 
+SELECT trip_id, start_time::timestamp, end_time::timestamp, EXTRACT(DAY FROM start_time::date) as start_time_day, EXTRACT (DAY FROM end_time::date) as end_time_day
+ FROM indego_trips_2020_q2
+  )
+SELECT count (*)
+FROM daytbl
+WHERE start_time_day != end_time_day
```

Comment:
Right track, but slightly over-counting, because some trip start and end times are more than one day apart (even though the max duration is minutes seconds). To really use this method, you would have to check for month boundaries as well (e.g. April 30 - May 1). The solution I used is [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L16-L25) (or a more compact equivalent solution [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L40-L42)). You could also [extract different parts from the date](https://www.postgresql.org/docs/9.1/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT), like `doy` (day of the year).